code/espurna/garland/animations/color_wave.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+enum WaveType {
+    Wave,
+    Comet
+};
+
+class ColorWave {
+   public:
+    ColorWave() = default;
+    /*
+        _numLeds - number of leds in garland
+        _palette - palette for wave colors
+        _waveLen - wave length in leds (for clean colors - length of one color), also fade length
+        _cleanColors - if true - use clean colors from palette, else - use interpolated colors
+        _fade - wave fade deep: 0 - no fade, 255 - fade to black
+        _pixelCache - pointer to array of Color with numLeds size, if not null - use it for caching colors
+        _speed - wave moving speed (0.5 - 2.0)
+        _dir - moving direction (1 or -1)
+        _startEmpty - if true - start wave from empty leds, else - start from full leds
+    */
+    ColorWave(
+        uint16_t _numLeds,
+        Palette *_palette,
+        uint16_t _waveLen,
+        bool _cleanColors,
+        byte _fade = 0,
+        Color* _pixelCache = nullptr,
+        WaveType _type = WaveType::Wave,
+        float _speed = secureRandom(5, 15) / 10.0,
+        int _dir = randDir(),
+        bool _startEmpty = false)
+        : numLeds(_numLeds), palette(_palette), waveLen(_waveLen), cleanColors(_cleanColors), fade(_fade), pixelCache(_pixelCache), type(_type), speed(_speed), dir(_dir) {
```

Comment:
(maybe ot)

just a reminder, named fields are a thing even in gcc4.8 (but only if one doesn't add default values)
Keeps ctors arglist from growing, imo keeps it more readable.
```cpp
struct Params {
  int a;
  int b;
};

...

class Bar {
  Bar(Params);
private:
  Params _params;
};

...

Bar(Params{ .a = 1, .b = 2 });
```

Reply:
Agree, the code is more clear now.